airqo-platform/AirQo-api
# query Events by timestamps

# query Events by timestamps (ISO 8601 format)

**_WHAT DOES THIS PR DO?_**

- [x] Resolves [issue-391](https://github.com/airqo-platform/AirQo-api/issues/391)
- [x] Modifies the query parameters for the Get Events endpoint. `startTime` and `endTime` can now accept both _days_ like `2021-02-14` and _timestamps_ like `2021-02-26T17:01:40.000Z`. 

**_WHAT JIRA STORY/TASK IS RELATED TO THIS PR?_**
N/A

**_WHAT IS THE LINK TO THE PR BRANCH_**
[issue-391](https://github.com/airqo-platform/AirQo-api/tree/issue-391)

**_HOW DO I TEST OUT THIS PR?_**
`cd AirQo-api/src/device-registry`
`npm install`
`npm run stage-mac` for Linux users OR `npm run stage-pc` for Windows.

Please ensure that your local Redis server is already running in a separate tab.


**_WHICH ENDPOINTS SHOULD BE READY FOR TESTING?:_** 

- Just the one for getting Events/measurements. Query values using `starTime` and/or `endTime`. Default: UTC current time.
- Just in case you are not interested in the most recent values, please ensure that you set the` recent` query to `no` since the default is always the most recent values. Also important to note that the device key in the API response changes from `_id` to` device` whenever the recent parameter is set to `no`.

Examples: 

- [GET] http://localhost:3000/api/v1/devices/events?tenant=airqo&startTime=2021-02-26T17:00:00.000Z&endTime=2021-02-26T18:00:00.000Z&device={DEVICE_NAME}&recent=no
- [GET] http://localhost:3000/api/v1/devices/events?tenant=airqo&endTime=2021-04-14&startTime=2021-02-14
- [GET] http://localhost:3000/api/v1/devices/events?tenant=airqo&endTime=2021-04-14
- [GET] http://localhost:3000/api/v1/devices/events?tenant=airqo&startTime=2021-02-14
-  [GET] http://localhost:3000/api/v1/devices/events?tenant=airqo&startTime=2021-02-26T17:01:40.000Z
- [GET]  [GET] http://localhost:3000/api/v1/devices/events?tenant=airqo&endTime=2021-02-26T18:00:00.000Z&device={DEVICE_NAME}&recent=no

**_IS THERE ANY JENKINS CONSOLE LINK FOR CODE COVERAGE AND BUILD INFO?_**
N/A

**_ARE THERE ANY RELATED PRs?_**
N/A
